### DISCUSSION TITLE
How can I add a context menu to ag grid?

### DISCUSSION THREAD

**Discussion by williwehrlos** [zauberzeug/nicegui#1376] [Q&A]
### Question
Edit: Just saw that it is an enterprise feature...

I saw that you can add a custom context menu to ag grid:

https://www.ag-grid.com/javascript-data-grid/context-menu/

Is this possible with nicegui?

**Comment by falkoschindler** [comment]
@williwehrlos I'm not an expert when it comes to AG Grid. Do you think a solution for context menus would still be possible and/or worthwhile? Or is the question obsolete because it's an enterprise feature?

**Comment by williwehrlos** [comment]
As you can use this Feature as demo in development I think it would be nice to have at least a proof of concept for this kind of extension.
The other thing I'm really interested in was mentioned here: https://github.com/zauberzeug/nicegui/discussions/510 

I would really like to add icons to each row to allow for calling funtions/links with ag grid component

**Reply by xaptronic** [reply]
I created an extension to support using aggrid enterprise features - I will try to make some examples of how to do various things including context menus and edit/view buttons.

https://github.com/xaptronic/nicegui-aggrid-enterprise
